spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/isolated/RepositoryReturnTypesIntegrationTests.java

```diff
@@ -0,0 +1,380 @@
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnBoxedDoubleFromInteger() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setBoxedInteger(Integer.MAX_VALUE);
+		allPossibleTypesRepository.save(entity);
+
+		Double result = allPossibleTypesRepository.findDoubleFromIntegerById(entity.getId());
+		assertThat(result, is(closeTo(entity.getBoxedInteger(), 0.01d)));
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnBoxedBoolean() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setBoxedBoolean(true);
+		allPossibleTypesRepository.save(entity);
+
+		Boolean result = allPossibleTypesRepository.findBoxedBooleanById(entity.getId());
+		assertThat(result, is(equalTo(entity.getBoxedBoolean())));
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnDate() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setDate(LocalDate.fromDaysSinceEpoch(1));
+		allPossibleTypesRepository.save(entity);
+
+		LocalDate result = allPossibleTypesRepository.findLocalDateById(entity.getId());
+		assertThat(result, is(equalTo(entity.getDate())));
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnTimestamp() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setTimestamp(new Date(1));
+		allPossibleTypesRepository.save(entity);
+
+		Date result = allPossibleTypesRepository.findTimestampById(entity.getId());
+		assertThat(result, is(equalTo(entity.getTimestamp())));
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnBigDecimal() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setBigDecimal(BigDecimal.ONE);
+		allPossibleTypesRepository.save(entity);
+
+		BigDecimal result = allPossibleTypesRepository.findBigDecimalById(entity.getId());
+		assertThat(result, is(equalTo(entity.getBigDecimal())));
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnBigInteger() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setBigInteger(BigInteger.ONE);
+		allPossibleTypesRepository.save(entity);
+
+		BigInteger result = allPossibleTypesRepository.findBigIntegerById(entity.getId());
+		assertThat(result, is(equalTo(entity.getBigInteger())));
+	}
+
+	/**
+	 * @see DATACASS-271
+	 */
+	@Test
+	public void shouldReturnEntityAsMap() {
+
+		AllPossibleTypes entity = new AllPossibleTypes("123");
+		entity.setPrimitiveInteger(123);
+		entity.setBigInteger(BigInteger.ONE);
+		allPossibleTypesRepository.save(entity);
+
+		Map<String, Object> result = allPossibleTypesRepository.findEntityAsMapById(entity.getId());
+		assertThat(result.size(), is(27));
+		assertThat(result.get("primitiveinteger"), is(equalTo((Object) Integer.valueOf(123))));
+		assertThat(result.get("biginteger"), is(equalTo((Object) BigInteger.ONE)));
+	}
+
+	public interface AllPossibleTypesRepository extends CrudRepository<AllPossibleTypes, String> {
```

Comment:
Not sure I follow. The Cassandra `timestamp` data type is tested with `shouldReturnTimestamp`.